org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultGradleBuild.java

```diff
@@ -46,7 +49,17 @@ public DefaultGradleBuild(BuildConfiguration buildConfig) {
     @Override
     public void synchronize(NewProjectHandler newProjectHandler, CancellationTokenSource tokenSource, IProgressMonitor monitor) throws CoreException {
         SynchronizeGradleBuildsOperation syncOperation = SynchronizeGradleBuildsOperation.forSingleGradleBuild(this, newProjectHandler);
-        syncOperation.run(tokenSource, monitor);
+        try {
+            syncOperation.run(tokenSource, monitor);
+            GradleMarkerManager.clear(this);
```

Comment:
❌ We should clear markers before, not after. We might for instance want to add warning markers which shouldn't be cleaned up.

Reply:
I was thinking about the same thing, although I ended up calling `clear()` after the synchronization because if the synchronization takes longer then the user won't see the errors for a while. It's just a thought though, I have no problem moving the `clear()` before the synchronization.

Reply:
I've moved `clear()` before the project synchronization.